src/test/scala/uk/gov/nationalarchives/tdr/api/service/ReferenceGeneratorServiceSpec.scala

```diff
@@ -0,0 +1,47 @@
+package uk.gov.nationalarchives.tdr.api.service
+
+import com.typesafe.config.ConfigFactory
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.client3.SimpleHttpClient
+import sttp.client3.testing.SttpBackendStub
+import sttp.model.Method
+
+import scala.concurrent.ExecutionContext
+
+class ReferenceGeneratorServiceSpec extends AnyFlatSpec with Matchers {
+  implicit val executionContext: ExecutionContext = ExecutionContext.Implicits.global
+
+  case class User(id: String)
+
+  "getReferences" should "return a list of references" in {
+    val sttpBackendStub = SttpBackendStub.synchronous
+      .whenRequestMatches(_.uri.path.startsWith(List("intg", "counter")))
+      .thenRespond("""["REF1","REF2"]""")
+    val referenceGeneratorService = new ReferenceGeneratorService(ConfigFactory.load, SimpleHttpClient(sttpBackendStub))
+    val getReferences = referenceGeneratorService.getReferences(2)
+
+    getReferences shouldBe List("REF1", "REF2")
+  }
+
+  "getReferences" should "return a list of references if referenceLimit is exceeded" in {
```

Comment:
Rename test slightly: `... return the correct number of reference if ''referenceLimit'' is exceeded in the request`